lib/inspec/resources/platform.rb

```diff
@@ -71,31 +71,26 @@ def params
     end
 
     def supported?(supports)
-      return true if supports.nil? || supports.empty?
-
-      status = true
-      supports.each do |support|
-        support.each do |k, v|
-          status =
-            case k
-            when :os_family, :"os-family", :platform_family, :"platform-family" then
-              in_family?(v)
-            when :os, :platform then
-              platform?(v)
-            when :os_name, :"os-name", :platform_name, :"platform-name" then
-              name == v
-            when :release then
-              check_release(v)
-            else
-              false
-            end
-
-          break if status == false
-        end
-        return true if status == true
-      end
+      raise ArgumentError, "`supports` is nil." unless supports
+
+      supports.any? { |support|
+        support.all? { |k, v|
+          case normalize(k)
+          when :os_family, :platform_family then
+            in_family?(v)
+          when :os, :platform then
+            platform?(v)
+          when :os_name, :platform_name then
+            name == v
+          when :release then
+            check_release(v)
+          end
+        }
+      } || supports.empty?
```

Comment:
The string/symbol normalization is both regrettable and an improvement :-)

But for the logic, I think this loses a lot of clarity - keep in mind other people who are not you may need to read and understand this. Imagine a junior engineer reading the original logic versus this. What is that dangling `||supports.empty?` doing? Easily missed. 

The original is much more straightforward and less clever.  I think we want code that is easy to maintain by a broad range of engineers.

Reply:
> The original is much more straightforward and less clever

I wholeheartedly disagree. Using enumerable methods spells out intent much more clearly than managing logic via local variables w/ `next`, `break` and the like. As a result using higher level intention describing methods is more junior developer friendly than the original version. Objectively, the new code is 83% the complexity of the original (17.5/20.9).

As far as the hanging `empty?` clause, it is lamentable, but always I prefer to put the positive/main cases before the negative/edge cases whenever possible. I could try to format it slightly differently but I prefer this structure.